packages-exp/auth-exp/src/platform_browser/popup_redirect.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SDK_VERSION } from '@firebase/app-exp';
+import * as externs from '@firebase/auth-types-exp';
+import { isEmpty, querystring } from '@firebase/util';
+
+import { AuthEventManager } from '../core/auth/auth_event_manager';
+import { AuthErrorCode } from '../core/errors';
+import { OAuthProvider } from '../core/providers/oauth';
+import { assert } from '../core/util/assert';
+import { _generateEventId } from '../core/util/event_id';
+import { _getCurrentUrl } from '../core/util/location';
+import { _open, AuthPopup } from '../core/util/popup';
+import { ApiKey, AppName, Auth } from '../model/auth';
+import {
+  AuthEventType,
+  EventManager,
+  GapiAuthEvent,
+  GapiOutcome,
+  PopupRedirectResolver
+} from '../model/popup_redirect';
+import { _openIframe } from './iframe/iframe';
+
+/**
+ * URL for Authentication widget which will initiate the OAuth handshake
+ */
+const WIDGET_URL = '__/auth/handler';
+
+export class BrowserPopupRedirectResolver implements PopupRedirectResolver {
```

Comment:
shouldn't there be a redirect method on this?

Reply:
also add this to index.ts

Reply:
Re. redirect method: coming soon to a PR near you!

Reply:
I'm going to also push the index.ts bit off for a bit. We'll need to do an instantiation thing like we do with the persistence classes, and I'd like to try to make all of that more generic